src/framework/components/sprite/sprite-animation-clip.js

```diff
@@ -289,7 +289,7 @@ class SpriteAnimationClip extends EventHandler {
         asset.off('remove', this._onSpriteAssetRemove, this);
 
         // unbind atlas
-        if (asset.resource && asset.resource.atlas) {
+        if (asset.resource && !asset.resource.atlas) {
```

Comment:
We only set the event listener if `.atlas` is null (see https://github.com/playcanvas/engine/blob/main/src/framework/components/sprite/sprite-animation-clip.js#L303)

So it makes sense to only remove it if it is still null (ie the texture atlas hasn't been loaded yet)